core/src/tangram.cpp

```diff
@@ -548,4 +559,9 @@ void setupGL() {
     while (Error::hadGlError("Tangram::setupGL()")) {}
 }
 
+void runOnMainLoop(std::function<void()> _task) {
+    std::lock_guard<std::mutex> lock(m_tasksMutex);
+    m_tasks.emplace(std::move(_task));
```

Comment:
Shouldn't `m_tasks` be cleared on a new scene load or a context loss?

Currently that means, old scene references will still be available in previous scene's `TileBuilder` instances living in `m_tasks`.